sql/src/main/java/io/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -316,11 +297,98 @@ private static Grouping computeGrouping(
           dimensions,
           aggregations,
           havingFilter,
-          RowSignature.from(rowOrder, postProject.getRowType())
+          RowSignature.from(rowOrderAndPostAggregations.rowOrder, aggregateProject.getRowType())
       );
     }
   }
 
+  @Nullable
+  private SortProject computeSortProject(
+      PartialDruidQuery partialQuery,
+      PlannerContext plannerContext,
+      RowSignature sortingInputRowSignature,
+      Grouping grouping
+  )
+  {
+    final Project sortProject = partialQuery.getSortProject();
+    if (sortProject == null) {
+      return null;
+    } else {
+      final List<PostAggregator> postAggregators = grouping.getPostAggregators();
+      final OptionalInt maybeMaxCounter = postAggregators
+          .stream()
+          .mapToInt(postAggregator -> Integer.parseInt(postAggregator.getName().substring(1)))
+          .max();
+
+      final RowOrderAndPostAggregations rowOrderAndPostAggregations = computePostAggregations(
+          plannerContext,
+          sortingInputRowSignature,
+          sortProject,
+          maybeMaxCounter.orElse(-1) + 1 // 0 if max doesn't exist
+      );
+
+      return new SortProject(
+          sortingInputRowSignature,
+          rowOrderAndPostAggregations.postAggregations,
+          RowSignature.from(rowOrderAndPostAggregations.rowOrder, sortProject.getRowType())
+      );
+    }
+  }
+
+  private static class RowOrderAndPostAggregations
+  {
+    private final List<String> rowOrder;
+    private final List<Aggregation> postAggregations;
+
+    RowOrderAndPostAggregations(List<String> rowOrder, List<Aggregation> postAggregations)
+    {
+      this.rowOrder = rowOrder;
+      this.postAggregations = postAggregations;
+    }
+  }
+
+  private static RowOrderAndPostAggregations computePostAggregations(
```

Comment:
How about calling this method simply `create`?

Reply:
Would you tell me your thoughts in more detail? `create` sounds too broad and less intuitive.

Reply:
I was thinking that `RowOrderAndPostAggregations.create` is still pretty obvious as to what it does.